org.mixedrealitytoolkit.input/InteractionModes/NearInteractionModeDetector.cs

```diff
@@ -25,49 +28,67 @@ public class NearInteractionModeDetector : ProximityDetector
         /// colliders stopped being detected and update their buttons front plate and rounded
         /// rect if they are labeled as dynamic (based on proximity).
         /// </summary>
-        List<Collider> previouslyDetectedColliders = new List<Collider>();
+        private List<Collider> previouslyDetectedColliders = new List<Collider>();
 
         /// <inheritdoc />
         public override bool IsModeDetected()
```

Comment:
Also curious about your thoughts on https://github.com/MixedRealityToolkit/MixedRealityToolkit-Unity/pull/611#discussion_r1489845921

> I'm not sure this is the best method to fire these events from, as it _could_ be called multiple times in a single frame. It also overloads the method a bit based on its simpler description: "Determine whether the detector has detected the appropriate conditions."

Reply:
> Also curious about your thoughts on [#611 (comment)](https://github.com/MixedRealityToolkit/MixedRealityToolkit-Unity/pull/611#discussion_r1489845921)
> 
> > I'm not sure this is the best method to fire these events from, as it _could_ be called multiple times in a single frame. It also overloads the method a bit based on its simpler description: "Determine whether the detector has detected the appropriate conditions."

At an earlier version of this PR my original approach was to filter the number of events risen by storing unique (Collider + Interactor) duples on the PressableButton side that were starting/stopping a ProximityHover events and only rise any such event when the corresponding collections would add or remove a unique duple.  Such approach is lost in translation at this point after implementing other feedback.  Granted, there were multiple calls to attempt to register a new (Collider + Interactor) duple but that would be prevented if the collection already had one such instance.

Reply:
I still don't think you need to store the collider at all...just the interactable....NOT the Interactor.

Reply:
Correct.  That's a remnant from the initial approach, I was under the (wrong) impression that it was needed to uniquely identify which UIElement was receiving the hoverProximity event and when trying to access both the FrontPlate RawImage and the Canvas' CanvasRoundedRect, now I realize it is indeed not needed for just modifying the FrontPlate's RawImage.  I've removed it in https://github.com/MixedRealityToolkit/MixedRealityToolkit-Unity/pull/659/commits/460c96e37cfc4d3ad2b7bfb54cdc14d4e9ed733d